src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts

```diff
@@ -28,7 +28,7 @@ export class AppLauncherDependencyCheckService implements DependencyCheckService
    */
   validateProjectName(projectName: string): boolean {
     // allows only '-'
-    const pattern = /^[a-z][a-z0-9-]{3,63}$/;
+    const pattern = /^[a-zA-Z0-9][a-zA-Z0-9-_\s]{2,62}[a-zA-Z0-9]$/;
```

Comment:
can we update tests with based on pattern change?

Reply:
sure

Reply:
please check it now

Reply:
There should be a test for each of the ways the user should not enter a name. Please submit a new PR to cover those too.